pkgs/development/compilers/zig/setup-hook.sh

```diff
@@ -1,5 +1,6 @@
 # shellcheck shell=bash disable=SC2154,SC2086
 
+# shellcheck disable=SC2034
```

Comment:
## [SC2034](https://www.shellcheck.net/wiki/SC2034) foo appears unused. Verify it or export it.

Why does `zigDefaultFlagsArray` appear unused to shellcheck? It certainly _looks_ used below...